azure/scope/cluster.go

```diff
@@ -297,6 +297,36 @@ func (s *ClusterScope) GroupSpec() azure.ResourceSpecGetter {
 	}
 }
 
+// VnetPeeringSpecs returns the virtual network peering specs.
+func (s *ClusterScope) VnetPeeringSpecs() []azure.VnetPeeringSpec {
+	peeringSpecs := make([]azure.VnetPeeringSpec, 2*len(s.Vnet().Peerings))
+
+	for i, peering := range s.Vnet().Peerings {
+		// If vnet is not provided, default to the resource group of the new vnet
```

Comment:
Consider defaulting this in the webhook instead so it's persisted in the spec

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1alpha4/azurecluster_default.go#L73